### PR TITLE
Shipment Summary Worksheet - Actual Obligations PPM Weight 

### PR DESCRIPTION
**Pull request by chrisrcoles**
## Description

System populates PPM shipment weight responsibility in the actual obligations compartment. If the remaining entitlement exceeds that of the PPM net weight, then the PPM net weight will be displayed. However, if the remaining entitlement is less than that of the PPM net weight, then the remaining entitlement weight will be displayed on the SSW. 

## Setup

`make server_test`

### Test in the office client
`make server_run`
`make office_client_run`

1) Log into the office app and select a move with a PPM
2) Make sure that the PPM has a "Net Weight" in the Weights panel
2) Click Create Paperwork -> Click Download worksheet

### Test with cmd line tool
` go run cmd/generate_shipment_summary/main.go -move 0db80bd6-de75-439e-bf89-deaafa1d0dc8 -debug`

## Code Review Verification Steps
* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References
* [Pivotal story](https://www.pivotaltracker.com/story/show/164283390) 

## Screenshots
ssw-ppm-actual-weight

### CODE REVIEW

File: pkg/models/shipment_summary_worksheet.go [hunk 1]
```diff
@@ -284,11 +283,21 @@ func formatActualObligationAdvance(data ShipmentSummaryFormData) string {
...
+
+		if ppmWeight >= weightAllotment {
+			actualWeightObligation = weightAllotment
+		} else {
+			actualWeightObligation = ppmWeight
+		}
 	}
-	return FormatWeights(0)
+	return FormatWeights(actualWeightObligation)
 }
```
Comment (mkrump) [inline pkg/models/shipment_summary_worksheet.go]: I'm kind of partial to putting the guard statement up top and un-nesting the if block

```
func FormatActualObligationsWeight(totalWeightAllotment int, personallyProcuredMoves PersonallyProcuredMoves) string {
	if len(personallyProcuredMoves) < 1 || personallyProcuredMoves[0].NetWeight == nil {
		return FormatWeights(0)
	}
	ppmWeight := int(*personallyProcuredMoves[0].NetWeight)
	weightAllotment := int(totalWeightAllotment)
	if ppmWeight >= weightAllotment {
		return FormatWeights(weightAllotment)
	}
	return FormatWeights(ppmWeight)
}
```

File: pkg/models/shipment_summary_worksheet.go [hunk 2]
```diff
@@ -269,9 +269,8 @@ func FormatValuesShipmentSummaryWorksheetFormPage1(data ShipmentSummaryFormData)
 	page1.MaxObligationGCCMaxAdvance = FormatDollars(data.MaxObligation.MaxAdvance())
 
 	page1.ActualObligationGCC100 = FormatDollars(data.ActualObligation.GCC100())
-	page1.ActualWeight = formatActualWeight(data)
+	page1.ActualWeight = FormatActualObligationsWeight(data.TotalWeightAllotment, data.PersonallyProcuredMoves)
```
Comment (mkrump) [inline pkg/models/shipment_summary_worksheet.go]: I wonder if it makes sense to move this part into the data section and just format the result here. I think there is going to be some additional logic for hhgs in future stories and could help keep formatting / data section separate.

Comment (chrisrcoles) [reply pkg/models/shipment_summary_worksheet.go]: That's actually how I had it originally - `formatWeight(FormatActualObligationsWeight(data.TotalWeightAllotment, data.PersonallyProcuredMoves))`. I changed it to keep in line with the implementation style of other examples here that format. But no problem - I'll switch it back.

File: pkg/models/shipment_summary_worksheet.go (hunk 1 above)
Comment (chrisrcoles) [reply pkg/models/shipment_summary_worksheet.go]: I like this better especially given that the logic will only continue to grow. Nice suggestion.

**Pull request merged**